src/internal.c

```diff
@@ -12613,6 +12732,11 @@ int SendKexDhGexRequest(WOLFSSH* ssh)
     if (ret == WS_SUCCESS)
         ret = wolfSSH_SendPacket(ssh);
 
+    if (ret == WS_SUCCESS) {
```

Comment:
Should we handle WANT_WRITE cases (and other error cases) here and the other couple locations where expectMsgId is set only if wolfSSH_SendPacket returns success. This differs from the pattern after SendNewKeys where it is unconditionally set after the packet has been created and attempted to be sent.